realtime/setupSocketIO.js

```diff
@@ -66,8 +67,18 @@ function setupNamespace(io) {
     perspective.findAll()
     .then((objArr) => {
       if (objArr) {
-        objArr.forEach((o) => rtUtils.initializeNamespace(o, io));
-        resolve(io);
+        objArr.forEach((o) => rtUtils.initializePerspectiveNamespace(o, io));
+        room.findAll()
+        .then((rooms) => {
+          if (rooms) {
+            rooms.forEach((r) => rtUtils.initializeBotNamespace(r, io));
+            resolve(io);
+          } else {
+            const err = new ResourceNotFoundError();
+            err.resourceType = 'Room';
+            throw err;
+          }
+        });
```

Comment:
This way, if the perpsective.findAll throws an error the room namespaces will not be initialized. If this is not the intention, you might want to decouple the perspective and room initialization.